readme.md

```diff
@@ -248,7 +248,35 @@ Default: `'text'`
 
 The parsing method. Can be `'text'`, `'json'` or `'buffer'`.
 
-The promise also has `.text()`, `.json()` and `.buffer()` methods which sets this and the `resolveBodyOnly` option automatically.
+The promise also has `.text()`, `.json()` and `.buffer()` methods which return another Got promise returning the parsed body.\
+It's like setting the options to `{responseType, resolveBodyOnly: true}` but without affecting the main Got promise.\
+Example:
+
+```js
+(async () => {
+	const responsePromise = got(url);
+	const bufferPromise = responsePromise.buffer();
+	const jsonPromise = responsePromise.json();
+
+	const [response, buffer, json] = Promise.all([responsePromise, bufferPromise, jsonPromise]);
+	// `response` is an instance of Got Response
+	// `buffer` is an instance of Buffer
+	// `json` is an object
+})();
+```
+
+Or if you don't care about the Response object:
+
+```js
+(async () => {
+	const bufferPromise = got(url).buffer();
+	const jsonPromise = bufferPromise.json();
```

Comment:
Why are you calling `.json()` on the `bufferPromise`? That makes no sense to me.

Reply:
It's just an example. The example above contains a Request instance, this one does not. 

> Why

`jsonPromise` may throw, so people may want to save the buffer somewhere for debugging purposes.

Reply:
I feel like the above example is clear enough and this example only adds confusion and it's not really needed.

Reply:
Sure, will remove this.